### PR TITLE
Generic iemessage

### PR DESCRIPTION
**Pull request by mauro-fanton**

### CODE REVIEW

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/auth/EnrolmentRequest.scala
Comment (mpbostock) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/auth/EnrolmentRequest.scala:1]: Can this file please be renamed to Requests as other requests are in this file and it looks like this is just a file for the EnrolmentRequest

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IE818Message.scala [hunk 1]
```diff
@@ -29,18 +29,34 @@ case class IE818Message
   private val key: Option[String],
   private val namespace: Option[String]
 ) extends IEMessage {
-  override def localReferenceNumber: Option[String] = None
+  def localReferenceNumber: Option[String] = None
```
Comment (mpbostock) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IE818Message.scala:32]: Do we need a localReferenceNumber here if the message doesn't have it? could the lrnEquals method just return false?

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala [hunk 2]
```diff
@@ -19,8 +19,15 @@ package uk.gov.hmrc.excisemovementcontrolsystemapi.models.messages
 import scala.xml.NodeSeq
 
 trait IEMessage {
-  def localReferenceNumber: Option[String]
-  def getType: String
+  // def not val as will be evaluated straight away
```
Comment (mpbostock) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala:22]: Can these comments be removed please?

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala [hunk 3]
```diff
@@ -19,8 +19,15 @@ package uk.gov.hmrc.excisemovementcontrolsystemapi.models.messages
...
+//  def localReferenceNumber: Option[String]
+//  def consignorId: Option[String]
+  def consigneeId: Option[String]
+  def administrativeReferenceCode: Option[String]
+  def messageType: String
   def toXml: NodeSeq
+  def getErns: Set[String] //= Set(consignorId, consigneeId).flatten
```
Comment (mpbostock) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala:29]: Can this commented out code be removed please?

**Review by mpbostock — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IE818Message.scala (hunk 1 above)
Comment (mauro-fanton) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IE818Message.scala:32]: yes, I agree, we dont need it here. I will looke into it.

Comment (mauro-fanton) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IE818Message.scala]: done

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/auth/EnrolmentRequest.scala
Comment (mauro-fanton) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/auth/EnrolmentRequest.scala:1]: what about AuthRequest?

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala (hunk 2 above)
Comment (mauro-fanton) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala]: done

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala (hunk 3 above)
Comment (mauro-fanton) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/messages/IEMessage.scala]: done

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/auth/EnrolmentRequest.scala
Comment (ChrisCookOC) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/auth/EnrolmentRequest.scala:1]: guess we should either bring AuthRequest in or take the other requests out.

Comment (mpbostock) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/auth/EnrolmentRequest.scala:1]: I just meant that this file has a lot of requests in it but the file is called EnrolmentRequest, which suggests it's the only request in this file. If we call it "Requests" then its more obvious that there's multiple requests that live here

Comment (mauro-fanton) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/auth/EnrolmentRequest.scala:1]: yes, I meant call the file AuthRequests as they all used in the authentication and all velidate/authenticate the lrn, ern, etc. Requests may be too generic

**Review by mpbostock — APPROVED** [review]

**Pull request merged**